green/cmdline.py

```diff
@@ -93,8 +93,15 @@ def main(argv: Sequence[str] | None = None, testing: bool = False) -> int:
         temp_dir_for_tests = tempfile.mkdtemp()
         atexit.register(lambda: shutil.rmtree(temp_dir_for_tests, ignore_errors=True))
         os.environ["TMPDIR"] = temp_dir_for_tests
+        prev_tempdir = tempfile.tempdir
         tempfile.tempdir = temp_dir_for_tests
-    return _main(argv, testing)
+        try:
+            return _main(argv, testing)
+        finally:
+            del os.environ["TMPDIR"]
```

Comment:
Ideally we would want reset TMPDIR to the previous version if any. Since it was the pre-existing behavior I'm approving since the changes are definitely a net improvement.

Thank you!

Reply:
We checked that TMPDIR was not set before entering this if branch. (Technically, it was either unset or set to an empty string, but these are the same for python's [tempfile logic](https://github.com/python/cpython/blob/af3c1d817d3f8369f8003965d967332a3a721a25/Lib/tempfile.py#L165), so I think it's fine to delete in either case.)